src/frequenz/sdk/data_ingestion/resampling/subscriptions.py

```diff
@@ -0,0 +1,40 @@
+"""
+Common model definitions used for resampling purposes.
+
+Copyright
+Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+
+from frequenz.channels import Receiver, Sender
+
+from frequenz.sdk.core.timeseries import Sample, TimeSeriesId
+
+
+@dataclass(frozen=True)
+class Subscription:
+    """Subscription for receiving data for a specific time series id."""
+
+    time_series_id: TimeSeriesId
+    start_time: Optional[datetime] = None
+
+
+@dataclass(frozen=True)
+class SubscriptionResponse:
+    """Response to a subscription request with a receiver for ingesting data."""
+
+    receiver: Receiver[Optional[Sample]]
+    subscription: Subscription
```

Comment:
Because the subscription holds the information about `TimeSeriesId`, so when you receive `SubscriptionResponse` in e.g. `BatteryPool` you know exactly which `(component_id, metric_id)` the `receiver` will be sending data for and from what `start_time`.

For instance, in `battery_pool.py`:
```
...
if msg := select.subscription_receiver:
    subscription_response = msg.inner
    receivers[subscription_response.subscription.time_series_id] = subscription_response.receiver
    # subscription_response.subscription.start_time is also available
```

Otherwise, as a caller, you would just receive `Receiver[Sample]`. I think including which `TimeSeriesId` the `Receiver` refers to, makes the message more complete.

Do you have any other suggestion how this could be done differently?

Reply:
@leandro-lucarella-frequenz resolving this, because subscriptions will be implemented differently